spec/micro/README.md

```diff
@@ -0,0 +1,638 @@
+
+The goal of this document is to serve as a "micro-IBC" specification that will allow new ecosystems to implement a protocol that can communicate with fully implemented IBC chains using the same security assumptions.
+
+The micro-IBC protocol must have the same security properties as IBC, and must be completely compatible with IBC applications. It may not have the full flexibility offered by standard IBC.
+
+### Desired Properties
+
+- Light-client backed security
+- Unique identifiers for each channel end
+- Authenticated application channel (must verify that the counterparty is running the correct client and app parameters)
+- Applications must be mutually compatible with standard IBC applications.
+- Must be capable of being implemented in smart contract environments with resource constraints and high gas costs.
+
+### Specification
+
+### Light Clients
+
+The light client module can be implemented exactly as-is with regards to its functionality. It **must** have external endpoints for relayers (off-chain processes that have full-node access to other chains in the network) to initialize a client, update the client, and submit misbehaviour in case the trust model of the client is violated by the counterparty consensus mechanism (e.g. committing to different headers for the same height).
+
+The implementation of each of these endpoints will be specific to the particular consensus mechanism targetted. The choice of consensus algorithm itself is arbitrary, it may be a Proof-of-Stake algorithm like CometBFT, or a multisig of trusted authorities, or a rollup that relies on an additional underlying client in order to verify its consensus.
+
+Thus, the endpoints themselves should accept arbitrary bytes for the arguments passed into these client endpoints as it is up to each individual client implementation to unmarshal these bytes into the structures they expect.
+
+```typescript
+// initializes client with a starting client state containing all light client parameters
+// and an initial consensus state that will act as a trusted seed from which to verify future headers
+function createClient(
+    clientState: bytes,
+    consensusState: bytes,
+): (Identifier, error)
+
+// once a client has been created, it can be referenced with the identifier and passed the header
+// to keep the client up-to-date. In most cases, this will cause a new consensus state derived from the header
+// to be stored in the client
+function updateClient(
+    clientId: Identifier,
+    header: bytes,
+): error
+
+// once a client has been created, relayers can submit misbehaviour that proves the counterparty chain
+// The light client must verify the misbehaviour using the trust model of the consensus mechanism
+// and execute some custom logic such as freezing the client from accepting future updates and proof verification.
+function submitMisbehaviour(
+    clientId: Identifier,
+    misbehaviour: bytes,
+): error
```

Comment:
As in, terminate the client with the reason? I guess the question is how much misbehavior should be tolerated. If it's zero than perhaps it would make sense to change the client state vs accumulate evidence of behavior for a later action.